### PR TITLE
Issue #112 hamburger

### PR DESCRIPTION
**Pull request by kutagorn**
I added antd [drawer](https://ant.design/components/drawer) to our mobile menu. 

There was an issue with the submenu's in smaller screen's. I solved it with responsive CSS to adjust submenu positioning for smaller screens. 

There is an other solution with the usage of secondary drawer but after implementing that with the top menu I did not liked how it looked. I still have the file on my computer if anyone want's to take a look at that. I prefered this way tho.

### CODE REVIEW

File: src/app/globals.css [hunk 1]
```diff
@@ -75,6 +75,13 @@ li:active,
     rgba(99, 4, 54, 1) 90%
   ) !important;
 }
+@media (max-width: 768px) {
+  .ant-menu-submenu-popup {
+    left: 0 !important;
```
Comment (ozgurozbek) [inline src/app/globals.css:80]: See, that breaks if screen is wider than submenu+menu width.

image

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -132,42 +165,56 @@ export default function Navbar() {
...
-          onClick={(e) => {
-            setselectedKeys([e.key]);
+
+        {/* Hamburger button for mobile */}
+        <Button
+          className="md:hidden"
+          type="text"
+          icon={<MenuOutlined />}
+          onClick={toggleDrawer}
+          style={{
```
Comment (ozgurozbek) [inline src/components/Navbar.tsx:175]: image

Not happy with the styling unfortunately. Would prefer it if it was larger and without the typical bordering etc.

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -107,20 +108,52 @@ const loadPageURL = (menuItem: any) => {
  */
 export default function Navbar() {
   const [selectedKeys, setselectedKeys] = useState<Array<string>>([]);
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
```
Comment (ozgurozbek) [inline src/components/Navbar.tsx:111]: Decent way to implement drawer honestly. If there was a built-in component that could be better utilized but the built in navbar is not that good anyway.

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -132,42 +165,56 @@ export default function Navbar() {
...
+          />
+        </Drawer>
+
+        {/* Normal menu (destop)*/}
+        <Menu
+          onClick={(e) => setselectedKeys([e.key])}
           selectedKeys={selectedKeys}
           theme="dark"
           mode="horizontal"
+          className="hidden md:flex"
```
Comment (ozgurozbek) [inline src/components/Navbar.tsx:210]: image

So what to do if that doesn't work 😂 See what's next to the logo.

File: src/components/Navbar.tsx [hunk 5]
```diff
@@ -107,20 +108,52 @@ const loadPageURL = (menuItem: any) => {
  */
 export default function Navbar() {
   const [selectedKeys, setselectedKeys] = useState<Array<string>>([]);
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
 
   useEffect(() => {
     setselectedKeys([loadPageURL(window.location.pathname.substring(1))]);
   }, []); // Empty dependency to make it run once
 
+  const menuItems = [
```
Comment (ozgurozbek) [inline src/components/Navbar.tsx:117]: Okay to move it outside if you want to call the menu from multiple places, but doing so in the same component and then calling it twice loads the whole menu twice.

image

**Review by ozgurozbek — CHANGES_REQUESTED** [review]
If you could address these issues I'd be happy to merge. Honestly, drawer approach is good if you could instead put [the inline menu from antd](https://ant.design/components/menu/#menu-demo-inline) inside it. Please please please make sure you aren't loading the same menu twice in the page. Navbar loads in every page, and is very crucial for navigation and performance.

**Comment by kutagorn** [comment]
Thanks for your efforts on this PR. After reviewing the changes, I've decided to close this PR and start fresh with a new branch. I ll try to do it without any bug's this time. This PR created more bug's then it solved.

**Pull request closed without merging**